app/scripts/modules/core/entityTag/entityTags.read.service.ts

```diff
@@ -34,8 +60,39 @@ export class EntityTagsReader {
       tag.lastModifiedBy = metadata.lastModifiedBy;
     }
   }
+
+  private collateEntityIds(entityType: string, entityIds: string[]): string[] {
+    const baseUrlLength = `${this.settings.gateUrl}/tags?entityType=${entityType}&entityIds=`.length;
+    const maxIdGroupLength = get(this.settings, 'entityTags.maxUrlLength', 4000) - baseUrlLength;
+    const idGroups: string[] = [];
+
+    if (entityIds.join(',').length > maxIdGroupLength) {
```

Comment:
I don't think it'll be too bad, since we'll only do it once more, and only if there are a lot of `entityIds` (and it's possible V8 is smart and caches it, anyway) - but it doesn't hurt to cache it so will do.